### PR TITLE
Fix consul session destroy waiting time

### PR DESCRIPTION
**Pull request by yanganto**

### CODE REVIEW

File: src/scoped_consul_session.rs [hunk 1]
```diff
@@ -46,6 +50,30 @@ impl<'a> ScopedConsulSession<'a> {
             wait = std::cmp::max(wait * 2, 5000);
         }
     }
+
+    /// Consumes and deletes consul session and block before the program die
+    pub fn destroy(self) {
+        let mut wait = 1;
```
Comment (timClicks) [inline src/scoped_consul_session.rs:56]:
```suggestion
        let mut wait = 1000;
```

File: src/consul_client.rs [hunk 2]
```diff
@@ -308,7 +308,7 @@ impl ConsulClient {
     }
 
     /// Delete a given consul session (`<https://www.consul.io/api-docs/session#delete-session>`)
-    pub async fn delete_session(&self, session: &ConsulSession) -> Result<()> {
+    pub async fn async_delete_session(&self, session: &ConsulSession) -> Result<()> {
```
[the remark this answers is not in the document]
Comment (yanganto) [reply src/consul_client.rs:311]: How does a second type work?
If the state change to `StateType::Shutdown`, we need the consul client to delete the session in a sync manner.  It still is the same instance and the same type.

**Comment by yanganto** [comment]
Hi @timClicks,
Does this PR look good?
If there is something I better change and I am not really got your idea, please kindly let me know.

**Review by Mic92 — APPROVED** [review]

**Comment by yanganto** [comment]
We do not really need these two destroy, and the sync one is removed.

File: src/consul_client.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (yanganto) [reply src/consul_client.rs]: We have a meeting and change some code of it.  Thanks for the review.

**Pull request merged**